### PR TITLE
New Endpoints

### PR DESCRIPTION
**Pull request by Mcray3000**
## shopping.py

**Babe, wake up. New endpoints just dropped.....**

### 1.4. Fulfill List - `/{user_id}/fulfill_list/{list_id}` (GET)

Generates a list of the closest stores that can fulfill the items 
in a shopping list, considering budget and maximum distance.

**Request:**

```json
{ 
  "budget": "integer",  // Optional: Maximum price per item in cents (default: 2147483647, greater than 0)
  "max_dist": "integer",  // Optional: Range in km (default: 10, greater than 0)
  "order_by": "integer" // Optional: Order by option (1=price,distance; 2=price; 3=distance, default: 1)
}
```
**Response**

```json
[
  {
    "Name": "string",          // Store name
    "Store ID": "integer",     // Store ID
    "Distance Away": "integer", // Distance to the store in km
    "Item": "string",          // Name of the food item
    "Price of Item": "integer"  // Price of the item in dollars
  },
  { 
    // ... more items
  }
]
```

**Possible Error Responses:**

*400 Bad Request:* 

Returned when the order_by query parameter is invalid.

```JSON
{
  "detail": "Invalid order_by option" 
}
```
*404 Not Found:*

Returned when the user, list, or association between them is not found,
or when no stores meet the specified criteria.

```JSON
{
  "detail": "User does not exist." 
}
```
```JSON
{
  "detail": "List does not exist." 
}
```
```JSON
{
  "detail": "User is not associated with this list." 
}
```
```JSON
{
  "detail": "No stores found with given parameters." 
}
```

### Find Snack - `/{user_id}/find_snack/{food_id}` (GET)

Finds the closest store with the specified food item for a given user, 
optionally considering the price and range.

**Request:**

```json
{ 
  "max_dist": "integer",  // Optional: Range in km (default: 10, greater than 0)
  "order_by": "integer"   // Optional: Order by option (1=price,distance; 2=price; 3=distance, default: 3)
}
```

**Response:**

```json
{
  "Name": "string",          // Store name
  "Store ID": "integer",     // Store ID
  "Distance Away": "float", // Distance to the store in km
  "Item": "string",          // Name of the food item
  "Price of Item": "integer"  // Price of the item in dollars
}
```

**Possible Error Responses:**

*400 Bad Request:*

Returned when the order_by query parameter is invalid.
```json
{
  "detail": "Invalid order_by option" 
}
```
*404 Not Found:* 

Returned when the user, food item, or store within the specified range is not found.

```json
{
  "detail": "User does not exist." 
}
```
```json
{
  "detail": "Food_id does not exist." 
}
```
```json
{
  "detail": "No stores in range." 
}
```

## Additional misc changes

- fixed cal poly longitude 
- simplified the optimize shopping route endpoint a bit

### CODE REVIEW

**Comment by TLMatthies** [comment]
How different is your new find snack endpoint from our existing optimize_shopping_route endpoint? Both check on the distance to stores based on a specific item. The only difference on the surface that I see is that optimize route can also include a user's budget and it returns multiple stores. If there is a big difference between the two, then I have no issue with adding this endpoint. Have yet to review the code though, getting to that now.

**Comment by Mcray3000** [comment]
Good note, Tim, they are very similar; I mostly made that one to replace the existing one (that's why so much of it is copied from the other one). Here are the key differences: 

- Use preferences are removed as we are not using that table anymore; it is now just an optional value
- It is a bit more specialized to return exactly what a user is looking for; a snack
     - Has 3 sorting options (get the closest store, get the cheapest store, and get the cheapest and closest store (if there's a tie)
- It has the in-range function, and it doesn't ask about the price because it's purely looking for the item you are asking for; you can say if you want the cheapest version, and it'll find what's available. The list one has more concerns about the budget. 
- The only other difference is the lists are now sorted within the database so it is not on the Python code, seems to run a little faster.

File: src/api/stores.py
Comment (TLMatthies) [inline src/api/stores.py:1]: I mean, I guess this is a change.

**Comment by TLMatthies** [comment]
Only thing that I have against the find snack endpoint is having the distance be a needed field. To get it to work I had to try and put 100000 km in. I'd much rather have it be a default of infinite range, only being a filter if the user inputs a number.
Same issue with fulfill list. Other than that, me likes.

**Review by TLMatthies — APPROVED** [review]
This is good for now, would like to see optional distances with default being no limit in the future.

**Pull request merged**